pallets/avn-anchor/src/mock.rs

```diff
@@ -0,0 +1,179 @@
+    type AccountId = AccountId;
+    type Lookup = IdentityLookup<Self::AccountId>;
+    type Block = Block;
+    type RuntimeEvent = RuntimeEvent;
+    type BlockHashCount = ConstU64<250>;
+    type Version = ();
+    type PalletInfo = PalletInfo;
+    type AccountData = pallet_balances::AccountData<Balance>;
+    type OnNewAccount = ();
+    type OnKilledAccount = ();
+    type SystemWeightInfo = ();
+    type SS58Prefix = ConstU16<42>;
+    type OnSetCode = ();
+    type MaxConsumers = ConstU32<16>;
+}
+
+parameter_types! {
+    pub const ExistentialDeposit: Balance = 1;
+}
+
+impl pallet_balances::Config for TestRuntime {
+    type MaxLocks = ConstU32<50>;
+    type Balance = Balance;
+    type RuntimeEvent = RuntimeEvent;
+    type DustRemoval = ();
+    type ExistentialDeposit = ExistentialDeposit;
+    type AccountStore = System;
+    type WeightInfo = ();
+    type MaxReserves = ();
+    type ReserveIdentifier = [u8; 8];
+    type FreezeIdentifier = ();
+    type MaxHolds = ();
+    type RuntimeHoldReason = ();
+    type MaxFreezes = ();
+}
+
+impl pallet_avn::Config for TestRuntime {
+    type RuntimeEvent = RuntimeEvent;
+    type AuthorityId = sp_runtime::testing::UintAuthorityId;
+    type EthereumPublicKeyChecker = ();
+    type NewSessionHandler = ();
+    type DisabledValidatorChecker = ();
+    type WeightInfo = ();
+}
+
+impl avn_proxy::Config for TestRuntime {
+    type RuntimeEvent = RuntimeEvent;
+    type RuntimeCall = RuntimeCall;
+    type Currency = Balances;
+    type Public = AccountId;
+    type Signature = Signature;
+    type ProxyConfig = TestAvnProxyConfig;
+    type WeightInfo = ();
+}
+
+impl Config for TestRuntime {
+    type RuntimeEvent = RuntimeEvent;
+    type RuntimeCall = RuntimeCall;
+    type Public = AccountId;
+    type Signature = Signature;
+    type WeightInfo = default_weights::SubstrateWeight<TestRuntime>;
+}
+// Test Avn proxy configuration logic
+#[derive(Copy, Clone, Eq, PartialEq, Ord, PartialOrd, Encode, Decode, Debug, TypeInfo)]
+pub struct TestAvnProxyConfig {}
+impl Default for TestAvnProxyConfig {
+    fn default() -> Self {
+        TestAvnProxyConfig {}
+    }
+}
+
+impl ProvableProxy<RuntimeCall, Signature, AccountId> for TestAvnProxyConfig {
+    fn get_proof(call: &RuntimeCall) -> Option<Proof<Signature, AccountId>> {
+        match call {
+            RuntimeCall::AvnAnchor(avn_anchor::Call::signed_register_chain_handler {
+                proof,
+                ..
+            }) |
+            RuntimeCall::AvnAnchor(avn_anchor::Call::signed_update_chain_handler {
+                proof, ..
+            }) |
+            RuntimeCall::AvnAnchor(avn_anchor::Call::signed_submit_checkpoint_with_identity {
+                proof,
+                ..
+            }) => Some(proof.clone()),
+            _ => None,
+        }
+    }
+}
+
+impl InnerCallValidator for TestAvnProxyConfig {
+    type Call = RuntimeCall;
+
+    fn signature_is_valid(call: &Box<Self::Call>) -> bool {
```

Comment:
Why do we need Box? Why not just `&Self::Call`?

Reply:
Because the size of Call is unknown (it can be any call to the runtime) and boxing it gives it a known size (the size of the pointer to the heap)